src/problem.ts

```diff
@@ -1 +1,62 @@
-export class Problem {};
\ No newline at end of file
+import { httpObject, Problem } from "types";
+
+enum COPY_MODE {
+    SKIP_PROPS = 'skipProps',
+    LEAVE_PROPS = 'leaveProps'
+}
+
+export class AsyncApiProblem extends Error implements Problem {
```

Comment:
We should call that class `Problem` and interface as `ProblemInterface` but we can change it in the next PRs.